tensorly/tenalg/einsum_tenalg/caching.py

```diff
@@ -0,0 +1,20 @@
+EINSUM_PATH_CACHE = dict()
+
+
+def einsum_path_cached(fun):
+    def wrapped(key, *args, **kwargs):
+        name = fun.__name__
+        try:
+            cache = EINSUM_PATH_CACHE[name]
+        except KeyError:
+            EINSUM_PATH_CACHE[name] = dict()
+            cache = EINSUM_PATH_CACHE[name]
```

Comment:
Is there a reason why you didn't use defaultdict here?

```suggestion
from collections import defaultdict


EINSUM_PATH_CACHE = defaultdict(dict)


def einsum_path_cached(fun):
    def wrapped(key, *args, **kwargs):
        name = fun.__name__
        cache = EINSUM_PATH_CACHE[name]
```

Reply:
That's a good idea - do you know how it does speed-wise? I was focused on trying to minimize any overhead. I guess it's not a big deal as that call will happen once anyway for each new fun.